pkg/webhooks/machine_webhook_test.go

```diff
@@ -3209,6 +3210,13 @@ func TestValidateGCPProviderSpec(t *testing.T) {
 			},
 			expectedOk: true,
 		},
+		{
+			testCase:         "with unknown fields in the providerSpec",
+			overrideRawBytes: []byte(`{"kind":"GCPMachineProviderSpec","apiVersion":"gcpprovider.openshift.io/v1beta1","metadata":{"creationTimestamp":null},"userDataSecret":{"name":"name"},"credentialsSecret":{"name":"name"},"canIPForward":false,"deletionProtection":false,"disks":[{"autoDelete":false,"boot":false,"sizeGb":16,"type":"","image":"","labels":null}],"networkInterfaces":[{"network":"network","subnetwork":"subnetwork"}],"serviceAccounts":[{"email":"email","scopes":["scope"]}],"machineType":"machineType","region":"region","zone":"region-zone","projectID":"projectID","gpus":[{"count":0,"type":"type"}],"onHostMaintenance":"Terminate","randomField-1": "something","randomField-2": "something"}`),
```

Comment:
It would be great if you could add a test, say for aws and azure, just to cover the main providerSpec providers.